### PR TITLE
#582: Use new XML merger with intellij configuration files

### PR DESCRIPTION
**Pull request by KianRolf**
fixes devonfw/IDEasy#582

Moves intellij config files from setup to update and configures them for the new xml merger

### CODE REVIEW

**Comment by KianRolf** [comment]
I am still not sure when exactly the OVERRIDE strategy is required

**Comment by KianRolf** [comment]
Still needs some adjustments

**Comment by KianRolf** [comment]
Should now be ready to be merged.

However some files are using override as merge strategy. This works for now but is not ideal.
The override merge strategy is used as a workaround to https://github.com/devonfw/IDEasy/issues/498.
Also i have discovered that the merge:id="name()" only works when the program runs in intellij (this is why the test are still green) but not in the finished build.
Screenshot 2024-11-13 155421

As a possible workaround i replaced the merge:id"name()" with xpathexpressions wich seemed to be working. So when #498 is resolved and the override merge strategies get removed i would recommend to try and use xpathexpressions

**Comment by hohwille** [comment]
https://stackoverflow.com/questions/65455908/graalvm-xml-issue

**Review by hohwille — CHANGES_REQUESTED** [review]
@KianRolf thanks for this PR. Quite tricky to add all this `merge` attributes and namespaces.
Thanks also for testing this with IDEasy and finding the problems in the XML merger. 👍 
I finally found the time to review - sorry for the long delay.
There are a lot of review comments I lefts.
Some of them are because of problems we already have before this PR so they are not your "fault" but I just noticed when I closely looked into them during the review.
Some problems would have been prevented by proper `git mv` usage so that should be a good learning for the future since git is fundamental for software development.
Please have a look and try to address the review comments.

File: intellij/workspace/update/.idea/workspace.xml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (KianRolf) [reply intellij/workspace/update/.idea/workspace.xml:1]: It works if the parent of MavenGeneralSettings gets the override merge:strategy

File: intellij/workspace/update/.intellij/config/options/windows/keymap.xml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
[the remark this answers is not in the document]
Comment (hohwille) [reply intellij/workspace/update/.intellij/config/options/windows/keymap.xml]: Was removed.

File: intellij/workspace/update/.intellij/config/options/jdk.table.xml [hunk 3]
```diff
@@ -1,13 +1,14 @@
...
-  <component name="ProjectJdkTable">
-    <jdk version="2">
-      <name value="Java" />
-      <type value="JavaSDK" />
-      <version value="OpenJDK $[JAVA_VERSION]" />
-      <homePath value="$[IDE_HOME]/software/java" />
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<application xmlns:merge="https://github.com/devonfw/IDEasy/merge" merge:strategy="override">
+  <component name="ProjectJdkTable" merge:id="@name">
+    <jdk version="2" merge:id="@version">
```
[the remark this answers is not in the document]
Comment (hohwille) [reply intellij/workspace/update/.intellij/config/options/jdk.table.xml:4]: Will close this conversation now since strategy is currently `override`.

**Pull request merged**